File: hw/jacome/U3/HW23Observer/HW23Observer/src/main/java/ec/edu/espe/hw23observer/model/Investor.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

package ec.edu.espe.hw23observer.model;

/**
 *
 * @author Micaela Jácome
 */
public class Investor implements IInvestor {
    private String name;

    public Investor(String name) {
        this.name = name;
    }

    public void update(Stock stock, Object args) {
        System.out.println("Notified observer " + name);
        if (args instanceof String) {
            System.out.println(args);
        } else if (args instanceof Double) {
            System.out.println("The price of " + stock.getSymbol() + " changed to: " + args);
        }
    }
}

File: hw/jacome/U3/HW23Observer/JS/model/Investor.js
const IInvestor = require('./IInvestor');

class Investor extends IInvestor {
    constructor(name) {
        super();
        this.name = name;
    }

    update(stock, args) {
        console.log(`Notified observer ${this.name}`);
        if (typeof args === 'string') {
            console.log(args);
        } else if (typeof args === 'number') {
            console.log(`The price of ${stock.getSymbol()} changed to: ${args}`);
        }
    }
}

module.exports = Investor;